File: RedSheet.DbEntities/Models/vTeamRole.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("vTeamRoles", Schema = "dbo")]
	[Serializable]
	public partial class vTeamRole  
	{
 
	[MaxLength(1000)]
	public string RoleDescription { get; set; } 
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int TeamRoleId { get; set; } 
 
	[Required] 
	[MaxLength(200)]
	public string TeamRoleName { get; set; } 

	public vTeamRole()
	{
    }

    }
}

File: redsheet-portal/src/app/database-models/v-team-role.ts
import { required, maxLength, range, nested } from '@rx/annotations';

export class vTeamRole {
    constructor(vTeamRole?: vTeamRole )  {
        let properties = [ "roleDescription", "teamRoleId", "teamRoleName",];
        for (let property of properties)
            if (vTeamRole && vTeamRole[property])
                this[property] = vTeamRole[property];
    }
 
    @maxLength(1000)
	roleDescription : string =   undefined;
 
	teamRoleId : number =   0 ;
 
    @required()
    @maxLength(200)
	teamRoleName : string =   undefined;


}
